src/main/java/org/apache/mesos/scheduler/plan/BlockFactory.java

```diff
@@ -11,5 +11,6 @@
  * detected for a Task.
  */
 public interface BlockFactory {
-    Block getBlock(TaskSpecification taskSpecification) throws InvalidRequirementException;
+    Block getBlock(String taskType, TaskSpecification taskSpecification)
```

Comment:
I'm having a hard time articulating it, but something feels off to me about the need now to pass just the name of the `TaskSet` around with the `TaskSpecification` all of the time. I don't think it should be plumbed down into `TaskSpecification` itself because it's obviously at a higher level of abstraction, but these two are now passed around as a pair frequently enough that I wonder if they comprise a latent abstraction that we're failing to see. Something to chew on.